### PR TITLE
Reorder command classes during interview

### PR DESCRIPTION
**Pull request by zacatac**
This has been tested against a number of devices and does seem to decrease interview flakiness. At the very least we are able to get the version of the version command class much earlier in the process.

### CODE REVIEW

File: node.go
Comment (parrotmac) [inline node.go]: super nit: typo

**Review by parrotmac — APPROVED** [review]

File: node.go
Comment (parrotmac) [inline node.go:265]: Wouldn't this contain both secure and insecure command classes?

File: node.go
Comment (parrotmac) [inline node.go:279]: Is it worth handling the case where there is a version command class but no security cc?

File: node.go
Comment (zacatac) [reply node.go:265]: No because we're iterating over the slice `insecure` which is initialized with the result of `ListBySecureStatus(false)`. The `ListBySecureStatus` function filters the support command classes for a given node and only returns the ones with the security status you specify i.e. true/false.

File: node.go
Comment (zacatac) [reply node.go:279]: I'd like to avoid doing that. I was definitely not able to pair if I interviewed the VersionCommandClass first, so I don't know if it's ever a good idea to hard code it to put that interview first.

Comment (parrotmac) [reply node.go:279]: Agreed we probably don't want it first for these devices, given that they require the security CC. However, it seems that this would break any device that _intentionally_ never reported the Security command class.

File: node.go
Comment (parrotmac) [reply node.go:265]: Ah, I understand now. Perfect.

File: node.go
Comment (zacatac) [reply node.go:279]: It wouldn't break it, it would just not attempt to reorder the command classes.

Comment (zacatac) [reply node.go:279]: There's a unit test for this case

Comment (parrotmac) [reply node.go:279]: Gotcha, makes sense :+1:

**Comment by zacatac** [comment]
Going to merge this once #38 has been merged.

**Pull request merged**
